src/App.jsx

```diff
@@ -14,70 +15,77 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setValue] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood === '' ? (
+          'No goods selected'
+        ) : (
+          `${selectedGood} is selected`
+        )}
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+        {selectedGood !== '' && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => {
+              setValue('');
+            }}
+          />
+        )}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={goods.indexOf(good)}
```

Comment:
The indexOf method is not considered the best practice for assigning keys in React because it returns the index of the first occurrence of an element in an array. If the array contains duplicate elements or if the order of the elements changes, it can result in incorrect keys being assigned to the elements.

Usually, developers use id for this purpose or different libraries to generate unique keys. In this task, it will be sufficient to use the product instead of the key, since we know that each product is unique 
```suggestion
              key={good}
```